### PR TITLE
STORM-658:when config topology.acker.executors is null, should not start acker bolts

### PR DESCRIPTION
**Pull request by caofangkun**
is null
Before the patch:
executors num is wrong

After the patch:

### CODE REVIEW

File: storm-core/src/clj/backtype/storm/daemon/common.clj [hunk 1]
```diff
@@ -312,7 +312,7 @@
     ))
 
 (defn has-ackers? [storm-conf]
-  (or (nil? (storm-conf TOPOLOGY-ACKER-EXECUTORS)) (> (storm-conf TOPOLOGY-ACKER-EXECUTORS) 0)))
+  (and (not (nil? (storm-conf TOPOLOGY-ACKER-EXECUTORS))) (> (storm-conf TOPOLOGY-ACKER-EXECUTORS) 0)))
```
Comment (harshach) [inline storm-core/src/clj/backtype/storm/daemon/common.clj]: can you use not-nil? in util.clj

Comment (d2r) [inline storm-core/src/clj/backtype/storm/daemon/common.clj]: Or leave out both `not` and `nil`?  Would that also work here?

Comment (caofangkun) [inline storm-core/src/clj/backtype/storm/daemon/common.clj]: Fixed. Thanks.

**Comment by ptgoetz** [comment]
-1. This breaks trident functionality if `topology.acker.executors` is `null` in `storm.yaml` and not overridden in the topology conf.

I've not dug too far into the root cause, but the situation above results in `StackOverflowError`s in the MasterBatchCoordinator spout. See stack trace below.
```
java.lang.StackOverflowError: null
        at clojure.lang.Numbers.multiply(Numbers.java:3663) ~[clojure-1.6.0.jar:na]
        at backtype.storm.stats$curr_time_bucket.invoke(stats.clj:29) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at backtype.storm.stats$update_rolling_window.doInvoke(stats.clj:41) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at clojure.lang.RestFn.applyTo(RestFn.java:142) ~[clojure-1.6.0.jar:na]
        at clojure.core$apply.invoke(core.clj:628) ~[clojure-1.6.0.jar:na]
        at backtype.storm.stats$update_rolling_window_set$iter__2980__2984$fn__2985$fn__2986.invoke(stats.clj:77) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at backtype.storm.stats$update_rolling_window_set$iter__2980__2984$fn__2985.invoke(stats.clj:76) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at clojure.lang.LazySeq.sval(LazySeq.java:40) ~[clojure-1.6.0.jar:na]
        at clojure.lang.LazySeq.seq(LazySeq.java:49) ~[clojure-1.6.0.jar:na]
        at clojure.lang.RT.seq(RT.java:484) ~[clojure-1.6.0.jar:na]
        at clojure.core$seq.invoke(core.clj:133) ~[clojure-1.6.0.jar:na]
        at clojure.core$dorun.invoke(core.clj:2855) ~[clojure-1.6.0.jar:na]
        at clojure.core$doall.invoke(core.clj:2871) ~[clojure-1.6.0.jar:na]
        at backtype.storm.stats$update_rolling_window_set.doInvoke(stats.clj:76) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at clojure.lang.RestFn.invoke(RestFn.java:439) ~[clojure-1.6.0.jar:na]
        at clojure.lang.Atom.swap(Atom.java:65) ~[clojure-1.6.0.jar:na]
        at clojure.core$swap_BANG_.invoke(core.clj:2234) ~[clojure-1.6.0.jar:na]
        at backtype.storm.stats$emitted_tuple_BANG_.invoke(stats.clj:215) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at backtype.storm.daemon.task$mk_tasks_fn$fn__4329.invoke(task.clj:160) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at backtype.storm.daemon.executor$fn__4615$fn__4630$send_spout_msg__4648.invoke(executor.clj:502) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at backtype.storm.daemon.executor$fn__4615$fn$reify__4657.emit(executor.clj:547) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at backtype.storm.spout.SpoutOutputCollector.emit(SpoutOutputCollector.java:49) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at storm.trident.topology.MasterBatchCoordinator.sync(MasterBatchCoordinator.java:176) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at storm.trident.topology.MasterBatchCoordinator.ack(MasterBatchCoordinator.java:145) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at backtype.storm.daemon.executor$ack_spout_msg.invoke(executor.clj:399) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at backtype.storm.daemon.executor$fn__4615$fn__4630$send_spout_msg__4648.invoke(executor.clj:532) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at backtype.storm.daemon.executor$fn__4615$fn$reify__4657.emit(executor.clj:547) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at backtype.storm.spout.SpoutOutputCollector.emit(SpoutOutputCollector.java:49) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at storm.trident.topology.MasterBatchCoordinator.sync(MasterBatchCoordinator.java:200) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at storm.trident.topology.MasterBatchCoordinator.ack(MasterBatchCoordinator.java:145) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at backtype.storm.daemon.executor$ack_spout_msg.invoke(executor.clj:399) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at backtype.storm.daemon.executor$fn__4615$fn__4630$send_spout_msg__4648.invoke(executor.clj:532) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at backtype.storm.daemon.executor$fn__4615$fn$reify__4657.emit(executor.clj:547) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at backtype.storm.spout.SpoutOutputCollector.emit(SpoutOutputCollector.java:49) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at storm.trident.topology.MasterBatchCoordinator.sync(MasterBatchCoordinator.java:176) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at storm.trident.topology.MasterBatchCoordinator.ack(MasterBatchCoordinator.java:145) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at backtype.storm.daemon.executor$ack_spout_msg.invoke(executor.clj:399) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
        at backtype.storm.daemon.executor$fn__4615$fn__4630$send_spout_msg__4648.invoke(executor.clj:532) ~[storm-core-0.10.0-SNAPSHOT.jar:0.10.0-SNAPSHOT]
```

**Comment by d2r** [comment]
This PR is a bit stale.  Can we move to another PR #628 that adds clarifying documentation, and close this one?

**Comment by HeartSaVioR** [comment]
@caofangkun Could you please close this PR? Thanks!

**Pull request closed without merging**